proc/threads.c

```diff
@@ -1275,138 +1402,309 @@ int threads_sigpost(process_t *process, thread_t *thread, int sig)
  */
 
 
-int _proc_lockSet(lock_t *lock, int interruptible, spinlock_ctx_t *sc)
+static unsigned int _proc_lockGetPriority(lock_t *lock)
 {
-	while (lock->v == 0) {
-		if (proc_threadWaitEx(&lock->queue, &lock->spinlock, 0, interruptible, sc) == -EINTR)
+	unsigned int priority = -1;
```

Comment:
We can safely start with e.g. 7 (sizeof ready list to avoid magic numbers) or thread->priority. -1 is somewhat misleading and can be dangerous if result of this function is used directly

Reply:
I've changed the starting lock priority to `sizeof(threads_common.ready) / sizeof(threads_common.ready[0]) - 1`. The function can be called on a lock without any waiting threads and should return lowest possible priority in that case.